File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>

bool is_armstrong_number(int candidate) {
    int nums[12];
    int buf = candidate;
    int sum = 0;
    int j = 0;
    while (buf > 0) {
        nums[j] = buf % 10;
        buf /= 10;
        ++j;
    }
    for (int i = 0; i < j; ++i)
        sum += pow(nums[i], j);
    return sum == candidate;
}

File: solutions/go/armstrong-numbers/1/armstrong_numbers.go
package armstrong

import "math"  

func IsNumber(n int) bool {
	nums := make([]int, 0)
    buf := n
    sum := 0
    for buf > 0 {
        nums = append(nums, buf % 10)
        buf /= 10
    }
    for i := 0; i < len(nums); i++ {
        sum += int(math.Pow(float64(nums[i]), float64(len(nums))))
    }
    return sum == n
}
